lightning/src/ln/chanmon_update_fail_tests.rs

```diff
@@ -1762,3 +1762,53 @@ fn during_funding_monitor_fail() {
 	do_during_funding_monitor_fail(true, true, true, true, false);
 	do_during_funding_monitor_fail(true, true, false, false, false);
 }
+
+#[test]
+fn test_path_paused_mpp() {
+	// Simple test of sending a multi-path payment where one path is currently blocked awaiting
+	// monitor update
+	let chanmon_cfgs = create_chanmon_cfgs(4);
+	let node_cfgs = create_node_cfgs(4, &chanmon_cfgs);
+	let node_chanmgrs = create_node_chanmgrs(4, &node_cfgs, &[None, None, None, None]);
+	let mut nodes = create_network(4, &node_cfgs, &node_chanmgrs);
+
+	let chan_1_id = create_announced_chan_between_nodes(&nodes, 0, 1, InitFeatures::supported(), InitFeatures::supported()).0.contents.short_channel_id;
+	let (chan_2_ann, _, chan_2_id, _) = create_announced_chan_between_nodes(&nodes, 0, 2, InitFeatures::supported(), InitFeatures::supported());
+	let chan_3_id = create_announced_chan_between_nodes(&nodes, 1, 3, InitFeatures::supported(), InitFeatures::supported()).0.contents.short_channel_id;
+	let chan_4_id = create_announced_chan_between_nodes(&nodes, 2, 3, InitFeatures::supported(), InitFeatures::supported()).0.contents.short_channel_id;
+
+	let (payment_preimage, payment_hash) = get_payment_preimage_hash!(&nodes[0]);
+	let payment_secret = PaymentSecret([0xdb; 32]);
+	let mut route = nodes[0].router.get_route(&nodes[3].node.get_our_node_id(), None, &[], 100000, TEST_FINAL_CLTV).unwrap();
```

Comment:
Worth asserting what route.paths[0] is, to make sure we're not just using 2 of the same path?

Reply:
pass_along_path does it for us, albeit indirectly.